### ISSUE TITLE
Framework Laptp 13 AMD Ryzen 7040 Series - cannot achieve Bitlocker hardware encryption with supported SSD

### ISSUE THREAD

**Issue by kennebnonedz** [FrameworkComputer/SoftwareFirmwareIssueTracker#24]
## Device Information
Framework Laptop 13 (AMD Ryzen™ 7040 Series) DIY 64GB RAM + 2.8k screen
Windows 11 Pro 23H2
AMD GPU driver 24.12.1

### System Model or SKU

Please select one of the following

- [ ] Framework Laptop 13 (11th Gen Intel® Core™)
- [ ] Framework Laptop 13 (12th Gen Intel® Core™)
- [ ] Framework Laptop 13 (13th Gen Intel® Core™)
- [x] Framework Laptop 13 (AMD Ryzen™ 7040 Series)
- [ ] Framework Laptop 13 (Intel® Core™ Ultra Series 1)
- [ ] Framework Laptop 16 (AMD Ryzen™ 7040 Series)

### BIOS VERSION
3.05
3.06 BETA

### DIY Edition information
Memory: Kingston FURY 64GB (2x32GB) 5600MHz CL40 Impact
Storage: Samsung 990 PRO 4TB

### Standalone Operation
Are you running your mainboard as a standalone device. Is standalone mode enabled in the BIOS?
- [ ] Yes
- [x] No

## Describe the bug
Unable to set DISABLE_BLOCK_SID in BIOS' storage security settings for Samsung 990 Pro in 'ready to encrypt' state - cannot achieve Bitlocker hardware encryption

Hello,
The issue is quite complicated and has been present for at least a year now - wider description can be found in Community thread https://community.frame.work/t/issues-enabling-bitlocker-hardware-encryption-windows-encrypted-hard-drive-on-amd-7840/39415 

Long story short - since the unlock password cannot be saved for the hardware encrypted drive (that's what DISABLE_BLOCK_SID is for), if drive is hardware encrypted, it's boot option is not visible for bios - Windows does not boot

## Steps To Reproduce
0. Get your Samsung SSD in ready to encrypt state (download Samsung Magician, update firmware etc. click on Data Management -> encrypted drive, toggle the setting)
1. Go to BIOS setup utility -> Security
2. 'Storage password setup page' menu is missing

## Expected behavior
'Storage password setup page' menu is present and storage password can be saved within the system

## Operating System (please complete the following information):
 - Windows 11 Pro
 - Version: 23H2
